geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/conf/AbstractArgResolver.scala

```diff
@@ -0,0 +1,94 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.conf
+
+import com.typesafe.config.ConfigParseOptions
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.annotation.tailrec
+
+/**
+ * Abstract trait for resolving sft/config parameters and handling errors around
+ * parsing.
+ */
+trait AbstractArgResolver[returnType, parseMethodArgTuple] extends LazyLogging {
+
+  // Object for holding relevant error to return
+  object ErrorData {
```

Comment:
instead of making this a global state, you could make it an instance class and pass it in through the `parseMethods` calls. that way, the methods will be thread-safe.
e.g.
```
class ErrorData { var message: String = _ ; var error: Throwable = _ }
...
val error = new ErrorData
parseMethods(args, None, parseMethodList, error))
...
 val e = new Throwable(error.message + "\n" + error.error.getMessage, error.error)